modules/percolator/src/test/java/org/elasticsearch/percolator/CandidateQueryTests.java

```diff
@@ -313,7 +314,7 @@ private BooleanQuery createRandomBooleanQuery(int depth, List<String> fields, Ma
                 numShouldClauses++;
             }
         }
-        builder.setMinimumNumberShouldMatch(numShouldClauses);
+        builder.setMinimumNumberShouldMatch(randomIntBetween(0, numShouldClauses));
```

Comment:
Did these changes to the randomized test catch the two bugs?

Reply:
I tried to make them more likely to catch bugs but unfortunately they did not.